### PR TITLE
[ticket/13497] Change get_tables() calls with $db_tools->sql_list_tables()

### PR DESCRIPTION
**Pull request by MGaetan89**
https://tracker.phpbb.com/browse/PHPBB3-13497
https://tracker.phpbb.com/browse/PHPBB3-9907

PHPBB3-13497

### CODE REVIEW

File: phpBB/includes/functions_compatibility.php [hunk 1]
```diff
@@ -195,3 +195,15 @@ function get_user_rank($user_rank, $user_posts, &$rank_title, &$rank_img, &$rank
 	$rank_img = $rank_data['img'];
 	$rank_img_src = $rank_data['img_src'];
 }
+
+/**
+ * Get tables of a database
+ *
+ * @deprecated 3.1.0 (To be removed: 3.3.0)
```
Comment (marc1706) [inline phpBB/includes/functions_compatibility.php]: Should be set to 3.1.3-RC1 right now.

**Comment by marc1706** [comment]
Tests are failing.

File: phpBB/includes/functions_compatibility.php (hunk 1 above)
Comment (MGaetan89) [inline phpBB/includes/functions_compatibility.php]: The function was already deprecated in 3.1.0. I just added the deprecated message and moved the function in `functions_compatibility.php`.

Comment (marc1706) [inline phpBB/includes/functions_compatibility.php]: Right, sorry about that.

**Comment by MGaetan89** [comment]
I will check that later.

**Comment by MGaetan89** [comment]
!unset WIP

**Comment by MGaetan89** [comment]
Tests are passing now

**Comment by nickvergessen** [comment]
Needs rebase.
This also fixes https://tracker.phpbb.com/browse/PHPBB3-9907 (adding to OP, so it's getting closed aswell)

**Comment by MGaetan89** [comment]
Rebased on `develop`.

File: phpBB/includes/functions_install.php [hunk 2]
```diff
@@ -276,7 +264,8 @@ function connect_check_db($error_connect, &$error, $dbms_details, $table_prefix,
 			$temp_prefix = strtolower($table_prefix);
 			$table_ary = array($temp_prefix . 'attachments', $temp_prefix . 'config', $temp_prefix . 'sessions', $temp_prefix . 'topics', $temp_prefix . 'users');
 
-			$tables = get_tables($db);
+			$db_tools = new \phpbb\db\tools\tools($db);
```
Comment (nickvergessen) [inline phpBB/includes/functions_install.php]: Please use the db\tools\factory instead of creating it directly

**Comment by MGaetan89** [comment]
Done.

File: phpBB/includes/functions_compatibility.php [hunk 3]
```diff
@@ -218,3 +218,15 @@ function get_remote_file($host, $directory, $filename, &$errstr, &$errno, $port
...
+ */
+function get_tables(&$db)
+{
+	$db_tools = new \phpbb\db\tools\tools($db);
```
Comment (nickvergessen) [inline phpBB/includes/functions_compatibility.php]: Factory

**Comment by MGaetan89** [comment]
Fixed.

**Comment by Nicofuma** [comment]
can you rebase please?

**Comment by MGaetan89** [comment]
Done.

**Comment by Nicofuma** [comment]
you need to rebase again :)

**Comment by MGaetan89** [comment]
Done :)

**Pull request merged**